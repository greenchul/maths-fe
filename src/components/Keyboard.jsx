const Keyboard = () => {
  return (
    <div>
      <input />
      <button>Submit</button>
    </div>
  );
};

export default Keyboard;
