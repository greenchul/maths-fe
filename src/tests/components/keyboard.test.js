import Keyboard from "../../components/Keyboard";
import { render, screen } from "@testing-library/react";

describe("keyboard component tests", () => {
  it("should render correctly", () => {
    const setAnswer = () => {};
    const { asFragment } = render(<Keyboard setAnswer={setAnswer} />);

    expect(asFragment()).toMatchSnapshot();
  });
  it("Input box should render correctly", () => {
    const setAnswer = () => {};
    render(<Keyboard setAnswer={setAnswer} />);
    const inputElement = screen.getByPlaceholderText("answer");

    expect(inputElement).toBeInTheDocument();
  });
});
