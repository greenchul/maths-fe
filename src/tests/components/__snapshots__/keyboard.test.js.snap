// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`keyboard component tests should render correctly 1`] = `
<DocumentFragment>
  <div>
    <input
      placeholder="answer"
      type="number"
    />
    <button>
      Submit
    </button>
  </div>
</DocumentFragment>
`;
